### ISSUE TITLE
Why do you need, duplicated handling for bn? 

### ISSUE THREAD

**Issue by sehyun03** [jwyang/faster-rcnn.pytorch#288]
```
  def train(self, mode=True):
    # Override train so that the training mode is set as we want
    nn.Module.train(self, mode)
    if mode:
      # Set fixed blocks to be in eval mode
      self.RCNN_base.eval()
      self.RCNN_base[5].train()
      self.RCNN_base[6].train()

      def set_bn_eval(m):
        classname = m.__class__.__name__
        if classname.find('BatchNorm') != -1:
          m.eval()

      self.RCNN_base.apply(set_bn_eval)
      self.RCNN_top.apply(set_bn_eval)
```

This code is from lib/model/faster_rcnn/resnet.py
You can handle train/eval mode for batch norm layer using "set_bn_eval" function.(just like below)
```
 def set_bn_eval(m):
        classname = m.__class__.__name__
        if classname.find('BatchNorm') != -1:
          m.eval()

      self.RCNN_base.apply(set_bn_eval)
      self.RCNN_top.apply(set_bn_eval)
```

However why did you used additional code for train/eval mode??
```
      self.RCNN_base.eval()
      self.RCNN_base[5].train()
      self.RCNN_base[6].train()
```

**Comment by 123zhen123** [comment]
I have the same question.Do you know it?
